torchtune/modules/transformer.py

```diff
@@ -449,6 +449,7 @@ def forward(
             if i in self.output_hidden_states:
                 hidden.append(h)
             # shape: [b, s, d]
+            torch._dynamo.mark_dynamic(h, 1)
```

Comment:
Noob question on where we apply `mark_dynamic`: is it here because this is the entry point for per-layer compile? (And separately would there be any benefit to compiling the token embeddings? I guess not?)

And similarly on logits and labels: because they are the entry point to the compiled loss?

Reply:
> is it here because this is the entry point for per-layer compile?

> And similarly on logits and labels: because they are the entry point to the compiled loss?

yes to both :)

> separately would there be any benefit to compiling the token embeddings? I guess not?

was thinking about it as well, and I just quickly tested it - seems no perf benefit from compiling the token embeddings :(